packages/tracing/src/transaction.ts

```diff
@@ -42,9 +43,9 @@ export class Transaction extends SpanClass implements TransactionInterface {
 
     this._trimEnd = transactionContext.trimEnd;
 
-    // _getNewTracestate only returns undefined in the absence of a client or dsn, in which case it doesn't matter what
-    // the header values are - nothing can be sent anyway - so the third alternative here is just to make TS happy
-    this.tracestate = transactionContext.tracestate || this._getNewTracestate() || 'things are broken';
+    this.metadata = transactionContext.metadata || {};
+    this.metadata.tracestate = this.metadata.tracestate || {};
+    this.metadata.tracestate.sentry = this.metadata.tracestate.sentry || this._getNewTracestate(this._hub);
```

Comment:
Is there a nicer/more compact way to do this? This feels like a lot of code, but I don't know of a setter version of the `?.` getter syntax.

Reply:
No, not really, but we do this in so many places, that I might actually write a `getOrDefault(this, 'metadata.tracestate.sentry', this._getNewTracestate(this._hub))` helper for this :<

Reply:
Like that https://lodash.com/docs/4.17.15#get

Reply:
Yes, except we also need to set the data. I think a helper isn't a bad idea, but I'll save that for a separate PR.